src/main/java/seedu/meal360/Parser.java

```diff
@@ -0,0 +1,641 @@
+                recipeList.editRecipe(recipeToEdit, ingredients);
+            }
+        } else if (index == 2) {
+            System.out.println("These are the ingredients for the recipe:");
+            ui.printSeparator();
+            Recipe recipe = parseViewRecipe(recipeName, recipeList);
+            ui.printRecipe(recipe);
+            ui.printSeparator();
+            System.out.println("Which ingredient do you want to change?");
+            int ingredientIndex = getInput.nextInt();
+            ingredientIndex -= 1;
+            int count = 0;
+            String ingredientToRemove = null;
+            for (String ingredient : recipeToEdit.getIngredients().keySet()) {
+                if (ingredientIndex == count) {
+                    ingredientToRemove = ingredient;
+                    System.out.println("Ingredient to be changed:");
+                    ui.printSeparator();
+                    String toPrint = String.format("%s(%d)", ingredient,
+                            recipeToEdit.getIngredients().get(ingredient));
+                    System.out.println(ui.formatMessage(toPrint));
+                    ui.printSeparator();
+                    break;
+                }
+                count++;
+            }
+            System.out.println("Please enter the new ingredient:");
+            String line = userInput.nextLine();
+            String command = line.replaceAll("\\s+", " ");
+            int indexOfEqual = command.indexOf("=");
+            String newIngredientName = command.substring(0, indexOfEqual);
+            int newIngredientQuantity = Integer.parseInt(command.substring(indexOfEqual + 1));
+            recipeToEdit.getIngredients().remove(ingredientToRemove);
+            recipeToEdit.getIngredients().put(newIngredientName, newIngredientQuantity);
+            recipeList.editRecipe(recipeToEdit, recipeToEdit.getIngredients());
+        } else if (index == 3) {
+            HashMap<String, Integer> newIngredientList = recipeToEdit.getIngredients();
+            System.out.println("These are the current ingredients:");
+            ui.printSeparator();
+            Recipe recipe = parseViewRecipe(recipeName, recipeList);
+            ui.printRecipe(recipe);
+            ui.printSeparator();
+            System.out.println(
+                    "Please Enter Additional Ingredients & Quantity (Enter done when complete): ");
+            while (true) {
+                String line = userInput.nextLine();
+                if (line.equals("done")) {
+                    ui.printSeparator();
+                    break;
+                } else {
+                    String[] command = line.trim().split(" ");
+                    ingredients = parseIngredientName(command);
+                    newIngredientList.putAll(ingredients);
+                }
+            }
+            recipeList.editRecipe(recipeToEdit, newIngredientList);
+        }
+
+        return recipeToEdit;
+    }
```

Comment:
Method is very large, consider taking out common behaviour and refactor to other methods for easier maintainability